File: 557. Reverse Words in a String III/557.solution.py
class Solution:
    def reverseWords(self, s: str) -> str:
      right = len(s) - 1
      reversedString = ""
      reversedWord = ""
      
      while (right >- 1):
        if (s[right] != " " or right == 0):
          reversedWord += s[right]
        
        if (s[right] == " " or right == 0):
          reversedString = reversedWord + reversedString
          if (right != 0):
            reversedString = " " + reversedString
          reversedWord = ""
          
        right -= 1
      
      return reversedString

File: 557. Reverse Words in a String III/557.solution.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function (s) {
  let right = s.length - 1;
  let reversedString = "";
  let reversedWord = "";
  while (right > -1) {
    if (s[right] !== " " || right === 0) reversedWord += s[right];

    if (s[right] === " " || right === 0) {
      reversedString = reversedWord + reversedString;
      if (right !== 0) reversedString = " " + reversedString;
      reversedWord = "";
    }

    right--;
  }
  return reversedString;
};
